src/views/product-integration/component-view/helpers/get-processed-variables-markdown.ts

```diff
@@ -0,0 +1,39 @@
+import { ReleaseComponent } from 'lib/integrations-api-client/release'
+import serializeIntegrationMarkdown from 'lib/serialize-integration-markdown'
+import { MDXRemoteSerializeResult } from 'next-mdx-remote'
+
+type ProcessedVariableMarkdown = { description?: MDXRemoteSerializeResult }
+export type ProcessedVariablesMarkdown = Record<
+	string,
+	ProcessedVariableMarkdown
+>
+/**
+ * Given a release component,
+ * which is expected to include `variable_groups`, each with `variables` which
+ * are expected to have properties that need markdown processing,
+ *
+ * Return a map of unique `variable` keys to processed markdown for
+ * each `variable`'s description. Processed markdown is stored as HTML.
+ */
+export async function getProcessedVariablesMarkdown(
+	releaseComponent: ReleaseComponent
+): Promise<ProcessedVariablesMarkdown> {
+	const processedVariablesMarkdown: ProcessedVariablesMarkdown = {}
+	const { variable_groups } = releaseComponent
+	for (const variableGroup of variable_groups) {
+		for (const variable of variableGroup.variables) {
+			const { key, description, variable_group_id } = variable
+			const uniqueKey = `${variable_group_id}.${key}`
+			processedVariablesMarkdown[uniqueKey] = {}
+			if (description !== null) {
+				// TODO: remove dev_markdown test values here
+				const dev_markdown = `**Test bold**, _emphasis_, also [a link](https://dev.hashicorp.com/not-visited-yet), and \`some inline code\`, heck maybe [\`linked code\`](https://dev.hashicorp.com/yoink)`
```

Comment:
I would also like to see what a code block looks like 🤔 

```sh
echo "something" # like this
echo "wow"
```

Reply:
Great question - I temporarily added it in to the dev value, here's a preview:

<img width="1439" alt="2023-02-02-code-block-demo" src="https://user-images.githubusercontent.com/4624598/216372351-1c29e473-12ac-49af-b264-d2cf0b100cc6.png">

Caveat: with the reduced "base" font size here compared to other long-form content (eg Docs and Learn), elements _other_ than the inline ones we've tweaked styles for in this PR won't match the reduced font-size used for rendering `description` content. We've flagged this previously with design during "alert" work (alerts also use `body-200` rather than the larger `body-300`) but might be worth flagging again if it seems like a priority to address this use case 🎏 

If we want to more fully account for "block" MDX components with this smaller `body-200` base size, I think we'd need a good chunk design involvement to figure out spacing and necessary tweaks to components such as code-block (unless we're happy with the differing relative font sizes & other metrics!). I've got a task set up with details: 🎟 [[discuss] smaller "body-200" mode for DevDotContent](https://app.asana.com/0/1202097197789424/1203869000261274/f)